monotonic/src/main/scala/com/criteo/cuttle/monotonic/MonotonicScheduler.scala

```diff
@@ -0,0 +1,165 @@
+package com.criteo.cuttle.monotonic
+
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+import com.criteo.cuttle.{Executor, Job, Scheduler, Scheduling, SchedulingContext, Workflow, XA, utils}
+import doobie.imports._
+import io.circe.{Encoder, Json}
+
+import scala.concurrent.Future
+import scala.concurrent.duration.{Duration => ScalaDuration}
+import scala.concurrent.stm._
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.reflect.ClassTag
+
+object MonotonicContext {
+  def apply[T: Encoder: ClassTag](execParams: T, execTime: LocalDateTime) = new MonotonicContext[T] {
+    def executionParams = execParams
+
+    def executionTime = execTime
+
+    val enc = implicitly[Encoder[T]]
+
+    override def toJson: Json = Json.obj(
+      "exec_params" -> enc(execParams),
+      "exec_time" -> Json.fromString(DateTimeFormatter.ISO_LOCAL_DATE_TIME.format(executionTime))
+    )
+
+    override def log: ConnectionIO[String] = Database.serializeContext(this)
+
+    override def compareTo(other: SchedulingContext): Int = other match {
+      case o: MonotonicContext[T] => ordering.compare(this, o)
+    }
+  }
+
+  implicit def ordering[T] = Ordering.fromLessThan((t1: MonotonicContext[T], t2: MonotonicContext[T]) => t1.executionTime.isBefore(t2.executionTime))
+}
+
+trait MonotonicContext[T] extends SchedulingContext {
+  /**
+    * Parameters of jobs.
+    * Each graph execution will provide different values.
+    *
+    * @return
+    */
+  def executionParams: T
+
+  /**
+    * Workflow execution time, used to order executions.
+    * This is the time of the completion of the trigger.
+    *
+    * @return graph execution time
+    */
+  def executionTime: LocalDateTime
```

Comment:
I'm using this to order executions and recreate a workflow from that execution time (to browse history of runs). This is the execution time of a workflow, not a job (I'll rename the func). I can add a UUID to act as a primary key but it may be useless.